src/Surfnet/ServiceProviderDashboard/Application/Command/Entity/SaveOidcngEntityCommand.php

```diff
@@ -358,6 +231,28 @@ public function __construct()
     {
     }
 
+    public function __set(string $property, ?Attribute $value)
```

Comment:
Should we allow setting null for an attribute value? You probably walked into this situation so you had to allow for this. But can you explain why?

It might be better to not actually set the value when a null-attribute is encountered. That shoudl also allow you to get rid of the ?Attribute return type on the `__get` and `getAttribute` methods.


An finally as a rule of thumb: methods starting with `get` should either return a result or throw an error when the result can not be found. Where find might result in a null result. I'm not saying you should change this just for the sake of convention. But I just wanted to share that knowledge.

Reply:
Thanks! On second thought I think a better way to use the __set and __get is to call a method the setAttribute and getAttribute and move the existence of the key in those methods, leaving the __get quite clean. I will refactor that in the next stage.